### DISCUSSION TITLE
About turtles and chunk loading

### DISCUSSION THREAD

**Discussion by vico93** [cc-tweaked/CC-Tweaked#1483] [General]
Well, i know @SquidDev stance is to not add it directly on CC:T, and the only other option that adds chunk loading to turtles also add a ton of things that i'm not interested (i really only want the chunk loading part).

But i was thinking, after seeing [this datapack](https://modrinth.com/datapack/chunk-loader): could a datapack be able to allow the turtle to load chunks on their own (without peripherals or equipped items)?

I would try to ask on Minecraft Commands server about it but before that i need to figure out what is, technically a turtle: a entity, a block, a Tile Entity... what? So i can pass the "context" to them to see if this is feasible.

**Comment by SquidDev** [comment]
Turtles are a block/block entity. I suspect it will not be possible to add chunk loaders to them via datapacks, as there's no way to track a turtle as it moves.

**Reply by vico93** [reply]
Yeah, i see. Someone on the Minecraft Commands server just confirmed it:
![image](https://github.com/cc-tweaked/CC-Tweaked/assets/3485505/5e978b66-8efd-4760-b44b-db2eaec0aaf7)
Unfortunately there is no options to keep loaded chunks on fabric 1.20 - there _is_ a mod that add kinda a chunk loader peripheral, but that feature is alongside a lot of others and changes that i dont have interest to.

For me just an addon mod that add an option (of by default if needed) to allow any turtles to keep chunks loaded is enough to me. Don't need to be an equippable item. Just the option.

**Reply by SquidDev** [reply]
I suspect your best bet is just to use one of the existing add on mods which adds this feature, and then remove the other items/blocks it adds via CraftTweaker/a data pack.

**Reply by SirEdvin** [reply]
Also, if this is about turtlematic, you can just disable everything (via configuration) else aside of chunk loader and modify recipe for chunk vial via regular datapack, probably.

**Comment by vico93** [comment]
> Also, if this is about turtlematic, you can just disable everything (via configuration) else aside of chunk loader and modify recipe for chunk vial via regular datapack, probably.

I'm not a fan of disabling ~90% of a large mod just to get a single one feature that could be implemented separately on an alternative addon mod for CC:T. For previous experience, doing similar things on large mods to get a few features always ended up in some conflict with other mods, aside of the extra unneeded size on the modpack (and it also requires dependencies...).

I will try to see if someone on Fabric discord have interest in make another addon mod that just alter the turtle behavior to support it loading chunks on their own.

**Comment by vico93** [comment]
After a while (and with a new MC update coming), i decided to come back here to see if SquidDev's stance about computer/turtle chunk loading changed at all (maybe reflecting possible internal changes that, in my ignorance, i could not be aware of and could facilitate things).

While trying to persuade SirEdvin about a standalone addon mod just for chunkloader he came up with a very interesting idea: adding custom gamerules that would allow whenever the turtles (and computers) should keep their chunk loaded (and, i think, load unloaded chunks they are heading in, in case of turtles) by default.

In a first glance that could generate a huge performance drop in big servers, but i really think that should be only used in singleplayer or small friend servers, so being disabled by default (where the wins of being able to keep chunks loaded by CC:T stuff overtakes the performance loss).

Unfortunately SirEdvin seems to be away for a while and this idea never materialized; i tried to see if someone at either MMD or Fabric's Discord would have interest, but seems that, since CC:T is a very niche mods, there are few people specialized on it to do such kind of addon mod.

**Reply by Lupus590** [reply]
You could have turtles send their location to a command computer which then force loads the chunk. It could also load the chunk in front of the turtle just before it crosses between 2 of them.

The hard part might be figuring out when you can unload a chunk. Perhaps turtles have to send a heartbeat of their location every minute or so to keep the chunk loaded.

**Reply by vico93** [reply]
> You could have turtles send their location to a command computer which then force loads the chunk. It could also load the chunk in front of the turtle just before it crosses between 2 of them.
> 
> The hard part might be figuring out when you can unload a chunk. Perhaps turtles have to send a heartbeat of their location every minute or so to keep the chunk loaded.

The issue is that it is a little bit "survival-breaking" to need to give a command computer. And if i'm not mistaken to setup gps is a kinda complicated to do at first (need more computers to triangulate the position)

**Reply by ArthurS1** [reply]
Any progress on this ? My turtles cannot work while I am away from my friends server :(
